core/src/main/java/com/netease/arctic/catalog/BaseIcebergCatalog.java

```diff
@@ -70,14 +72,24 @@ public void initialize(
 
   @Override
   public List<String> listDatabases() {
-    if (icebergCatalog instanceof SupportsNamespaces) {
-      return tableMetaStore.doAs(() -> ((SupportsNamespaces) icebergCatalog).listNamespaces(Namespace.empty()).stream()
-          .map(namespace -> namespace.level(0)).distinct().collect(Collectors.toList()));
-    } else {
+    if (!(icebergCatalog instanceof SupportsNamespaces)) {
       throw new UnsupportedOperationException(String.format(
           "Iceberg catalog: %s doesn't implement SupportsNamespaces",
           icebergCatalog.getClass().getName()));
     }
+
+    List<String> databases =
+        tableMetaStore.doAs(() ->
+            ((SupportsNamespaces) icebergCatalog).listNamespaces(Namespace.empty())
+                .stream()
+                .map(namespace -> namespace.level(0))
+                .distinct()
+                .collect(Collectors.toList())
+        );
+    String filter = meta.getCatalogProperties().get(CatalogMetaProperties.KEY_INCLUDE_DATABASES);
+    return databases.stream()
+        .filter(database -> filter == null || Pattern.matches(filter, database))
```

Comment:
Maybe we can add a class variable `databaseFilterPattern` and initialize it in method `initialize `, so we will not need to read these properties and construct Pattern every time?

Reply:
done.